README.md

```diff
@@ -1 +1,47 @@
-## Canvas Course Manager
\ No newline at end of file
+## Canvas Course Manager
+
+## Development
+
+Here are some basic set-up instructions.
+
+1. Build two separate images for server and client.
+    ```
+    docker-compose build
+    ```
+
+2. Start up the client and server applications in separate containers.
+    ```
+    docker-compose up
+    ```
+
+3. Access the client by visiting `http://localhost:4001` in your browser of choice.
+
+Use `^C` to stop the container and `docker-compose down` to remove the last used image from staging.
+
+Many JavaScript and template changes can be made without re-building the image,
+but as necessary, rebuild the image using Step 1.
+
+## Production
+
+Taken together, all the stages in `ccm_web/Dockerfile` will build an optimized image for production.
+
+You can build a production image, run a container, and use the application locally by performing
+the following steps, replacing `{PORT}` with the port you want the application to run on:
+
+1. Build one image with the server and static artifacts, including the bundled JavaScript frontend.
+
+```
+docker build -t ccm --build-arg PORT={PORT} ./ccm_web
+```
```

Comment:
I see the build failure for this case
```
TypeScript error in /build/client/src/App.tsx(1,19):
Could not find a declaration file for module 'react'. '/build/node_modules/react/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/react` if it exists or add a new declaration (.d.ts) file containing `declare module 'react';`  TS7016
```

Reply:
Uh-oh, let me take a look. I tested this, but I might have had some cache or directories disrupting the outcome.

Reply:
Reproduced. I guess some of those Typescript-related packages do need to be `dependencies`...

Reply:
Okay, I think I fixed this build problem here: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/24/commits/b5d981fdfce69048ce79f24bd4a4a77aafd34521

I'm looking into alternatives for development setups.

Reply:
With webpack and `ts-loader` handling the TypeScript transformations, I think we can keep all of the `@types` packages as `devDependencies`. I will test this more.